src/Swashbuckle.AspNetCore.SwaggerUI/SwaggerUIMiddleware.cs

```diff
@@ -67,7 +68,7 @@ public async Task Invoke(HttpContext httpContext)
                 return;
             }
 
-            if (httpMethod == "GET" && Regex.IsMatch(path, $"^/{Regex.Escape(_options.RoutePrefix)}/?index.html$",  RegexOptions.IgnoreCase))
```

Comment:
@WeihanLi the change to this line breaks the case where the `RoutePrefix` is empty, which is a legit config option.

We could add a conditional, which will further complicate the code, but now that I think about it - I'd be amazed if the previous implementation based on a simple Regex pattern was truly causing performance problems. Unless you have a better way to support the empty `RoutePrefix`, I'm tempted to revert this change (although I'd retain the async change below).

Reply:
Thanks @domaindrivendev , seemed missed a `Trim('/')`, would add a test case for this, thanks

Reply:
Hi @domaindrivendev , I've opened another PR to fix this https://github.com/domaindrivendev/Swashbuckle.AspNetCore/pull/2323